### PR TITLE
[share_plus] minSdkVersion 16 & graceful fallback

### LINKED ISSUE #789: share_plus Android minSdk bump drops support for a lot of devices
## System info

Platform the Issue occurs on: Android
Plugin name: share_plus
Plugin version: 4.0.1

## Steps to Reproduce

The latest `share_pus` plugin update bumped minSdk Android API from 16 to 23 (from Android 4.1 to Android 6). With Flutter itself supporting API 16 that is a large drop of supported Android devices.

My app is primarily used on Android tablets, and there is a lot of them running older Android versions. For example, older generation Amazon Fire 7 tablets (Fire OS 5 is API level 22) and some of 3..4 year old Samsung's Tab As are also running Android 5 or older.

### PR DESCRIPTION
**Pull request by Coronon**
## Description

- Revert `minSdkVersion` back to 16 and dynamically determine support for `shareWithResult` at runtime
- Gracefully fall back to "normal" share without result if the environment does not support `shareWithResult`
- Improve documentation for `shareWithResult`

## Related Issues

Closes #789

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by Coronon** [comment]
The conflicts are due to #797. I will resolve them tonight (mainly formatting).

**Comment by Coronon** [comment]
Did not think about that - yeah, that would definitely simplify the API, only problem I see is people not awaiting the "old" `share` methods and then complaining about the "not awaited..." error because lets face it: most people just update and don't read the changelog ^^

File: packages/share_plus/share_plus_platform_interface/CHANGELOG.md
Comment (vbuberen) [inline packages/share_plus/share_plus_platform_interface/CHANGELOG.md:3]:
```suggestion
- Gracefully fall back from `shareWithResult()` to regular `share()` methods on unsupported platforms
```

File: packages/share_plus/share_plus/lib/share_plus.dart
Comment (vbuberen) [inline packages/share_plus/share_plus/lib/share_plus.dart:95]: I think we should clearly state the fact that on Android and macOS we can't guarantee share action as it is something that definitely requested by people using `withResult` calls. 

```suggestion
  /// Android and macOS only record if the user selected an action or outright
  /// dismissed the share-sheet. It is not guaranteed that user actually shared something.
```

File: packages/share_plus/share_plus/lib/share_plus.dart
Comment (vbuberen) [inline packages/share_plus/share_plus/lib/share_plus.dart:124]: Same as above.
```suggestion
  /// Android and macOS only record if the user selected an action or outright
  /// dismissed the share-sheet. It is not guaranteed that user actually shared something.
```

**Review by vbuberen — COMMENTED** [review]
Left a few comments on namings.

What we are currently missing is updates to docs, which appear on Plus plugins website. 
Would be good if you also modify https://github.com/fluttercommunity/plus_plugins/blob/main/docs/share_plus/usage.mdx to mention new methods and description about fallback on unsupported platforms.

File: packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt
Comment (Coronon) [reply packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt:18]: `isWithResultAvailable` is a little confusing in itself as the result may be available but not requested

**Comment by Coronon** [comment]
Also added `shareWithResult` and `shareFilesWithResult` to the **usage.mdx** docs.

File: packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt
Comment (vbuberen) [reply packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt:18]: Ok, how about going a little simpler and making it `isWithResult`, considering its usage?

Comment (Coronon) [reply packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt:18]: `withResult` already sounds like a **bool** to me… I can change the name, but I would personally not consider that an improvement - the opposite rather

Comment (vbuberen) [reply packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt:18]: I don't agree here. Prefixing booleans with `is`, `has`, etc. is something that comes from Java times and is still widely used in Android development. It is something that helps the person reading the code to understand that it is boolean right away.

**Review by vbuberen — APPROVED** [review]
LGTM. Thanks for this contribution!
Would like to wait for a review from Miguel as he was the person, who started reviewing this PR.

**Comment by miquelbeltran** [comment]
LGTM too, I will go ahead and merge & release

**Comment by uc-asa** [comment]
I'm still getting this issue and android always return null in raw

**Pull request merged**
